### DISCUSSION TITLE
Feature Request: Plotting Helpers

### DISCUSSION THREAD

**Discussion by rymaju** [themetaschemer/malt#7] [Ideas]
First thing I wanted to do when I saw graphs in the book is make some myself using Racket's wonderful `plot` library. Sadly, its a bit cumbersome since behind the scenes, `1.0` is not a `number?`.

Some kind of helper that takes a dataset, target function, and guessed theta and displays the dataset as points and the target function would be ideal.

This is my current workaround for context:

```racket
;; line is already defined, from malt
(require (except-in plot line))

;; (U Tensor Scalar) -> (U List Number)
;; Convert the tensor/scalar into a list/number
(define (tensor->datum t)
  (cond [(scalar? t) (ρ t)]
        [else (map (curry tref t) (range (tlen t)))]))

;; Do the book!
(match-define θ
  (with-hypers
    ((revs 1000)
     (α 0.01)
     (batch-size 10))
    (gradient-descent
     (sampling-obj
      (l2-loss line) line-xs line-ys)
     (list 0.0 0.0))))

;; Target Parameters -> (Real -> Real)
;; Assumes the target function computes a scalar value.
(define (target->plottable-function target θ)
  (λ (x)
    (tensor->datum ((target x) θ))))

(plot (list
           (axes)
           (points (map list (tensor->datum line-xs) (tensor->datum line-ys)))
           (function (target->plottable-function line θ))))

```
![image](https://user-images.githubusercontent.com/23691775/221386706-c8dfdd3f-209b-4992-8e7d-f247906e1da6.png)

**Comment by themetaschemer** [comment]
Hi @rymaju Thanks for the suggestion. Your solution works fine. Adding built-in renderer support for tensor representations will definitely allow for some good plotting of many different things. I will take it into consideration for a future release.
